src/Tasks/FileIO/GetFileHash.cs

```diff
@@ -113,39 +123,13 @@ internal static string EncodeHash(HashEncoding encoding, byte[] hash)
         internal static bool TryParseHashEncoding(string value, out HashEncoding encoding)
             => Enum.TryParse<HashEncoding>(value, /*ignoreCase:*/ true, out encoding);
 
-        internal static bool SupportsAlgorithm(string algorithmName)
-            => _supportedAlgorithms.Contains(algorithmName);
-
-        internal static byte[] ComputeHash(string algorithmName, string filePath)
+        internal static byte[] ComputeHash(Func<HashAlgorithm> algorithmFactory, string filePath)
         {
             using (var stream = File.OpenRead(filePath))
-            using (var algorithm = CreateAlgorithm(algorithmName))
+            using (var algorithm = algorithmFactory())
```

Comment:
Can you replace these with `return algorithmFactory().ComputerHash(File.OpenRead(filePath));`? If so, then you might consider inlining it, since it's only used twice.

Reply:
Wouldn't you need to dispose the algorithm and the file stream instance? Changing it to your suggestion changes the semantic of that code.

Reply:
You're right.